app/views/catalog/_index_default.html.erb

```diff
@@ -5,13 +5,13 @@
   <% index_fields(document).each do |solr_fname, field| -%>
     <% if should_render_index_field? document, field %>
       <% if document[solr_fname].length == 1 || !document[solr_fname].is_a?(Array) %>
-        <% valdir = getdir(document[solr_fname])%>
+        <% valdir = document[solr_fname].first.dir %>
```

Comment:
How about just wrapping it in an Array to make sure first will always work?  Seems like less duplicate code that way:

`valdir = Array(document[solr_fname]).first.dir`